controller/dataplane/watch.go

```diff
@@ -79,3 +87,28 @@ func listDataPlanesReferencingKongPluginInstallation(
 		})
 	}
 }
+
+func listDataPlanesReferencingDataPlaneKonnectExtension(
+	c client.Client,
+) handler.TypedMapFunc[*operatorv1alpha1.DataPlaneKonnectExtension, reconcile.Request] {
+	return func(
+		ctx context.Context, ext *operatorv1alpha1.DataPlaneKonnectExtension,
+	) []reconcile.Request {
+		logger := ctrllog.FromContext(ctx)
+
+		// Find all DataPlane resources referencing DataPlaneKonnectExtension
+		// that maps to the DataPlaneKonnectExtension enqueued for reconciliation.
+		var dataPlaneList operatorv1beta1.DataPlaneList
+		if err := c.List(ctx, &dataPlaneList, client.MatchingFields{
+			index.DataPlaneKonnectExtensionIndex: ext.Namespace + "/" + ext.Name,
+		}); err != nil {
+			logger.Error(err, "Failed to list DataPlanes in watch", operatorv1alpha1.DataPlaneKonnectExtensionKind)
+			return nil
+		}
+		return lo.Map(dataPlaneList.Items, func(dp operatorv1beta1.DataPlane, _ int) reconcile.Request {
+			return reconcile.Request{
+				NamespacedName: client.ObjectKeyFromObject(&dp),
+			}
+		})
```

Comment:
We could use `objectListToReconcileRequests` but sadly that's in `konnect` package unexported now 😞 We should move that at some point to a watch dedicated package.

https://github.com/Kong/gateway-operator/blob/0d9d721bfca68961e6b7f6ab176f6feff8d5566e/controller/konnect/watch.go#L118-L140

Reply:
Honestly, I am not a big fan of using generics for everything. There are cases where it is definitely worth it but in cases such as this one, where 3 lines of easy code are replaced by 1, I don't know...

Reply:
Understood. I just want to point out that that these 3 lines (5 if we're precise) become:

```suggestion
		return objectListToReconcileRequests(dataPlaneList)
```

Which I believe is pretty elegant. Nonetheless we can leave it at what it is now 👍